sysboundary/ionosphere.cpp

```diff
@@ -1699,6 +1711,620 @@ namespace SBC {
+               MPI_Allreduce(cellBWTracingCoordinates.data(), sumCellBWTracingCoordinates.data(), 3*globalDccrgSize, MPI_FLOAT, MPI_SUM, MPI_COMM_WORLD);
+            }
+            anyCellNeedsTracing = false;
+         }
+         #pragma omp barrier
+         phiprof::stop("MPI-loop");
+         
+         #pragma omp for schedule(dynamic) reduction(||:anyCellNeedsTracing)
+         for(int n=0; n<globalDccrgSize; n++) {
+            if(sumCellNeedsContinuedFWTracing[n] > 0) {
+               anyCellNeedsTracing=true;
+               cellNeedsContinuedFWTracing[n] = 1;
+               
+               // Update that nodes' tracing coordinates
+               cellFWTracingCoordinates[n][0] = sumCellFWTracingCoordinates[n][0] / sumCellNeedsContinuedFWTracing[n];
+               cellFWTracingCoordinates[n][1] = sumCellFWTracingCoordinates[n][1] / sumCellNeedsContinuedFWTracing[n];
+               cellFWTracingCoordinates[n][2] = sumCellFWTracingCoordinates[n][2] / sumCellNeedsContinuedFWTracing[n];
+               
+               cellFWStepCounter[n] = maxCellFWStepCounter[n];
+            }
+            if(sumCellNeedsContinuedBWTracing[n] > 0) {
+               anyCellNeedsTracing=true;
+               cellNeedsContinuedBWTracing[n] = 1;
+               
+               // Update that nodes' tracing coordinates
+               cellBWTracingCoordinates[n][0] = sumCellBWTracingCoordinates[n][0] / sumCellNeedsContinuedBWTracing[n];
+               cellBWTracingCoordinates[n][1] = sumCellBWTracingCoordinates[n][1] / sumCellNeedsContinuedBWTracing[n];
+               cellBWTracingCoordinates[n][2] = sumCellBWTracingCoordinates[n][2] / sumCellNeedsContinuedBWTracing[n];
+               
+               cellBWStepCounter[n] = maxCellBWStepCounter[n];
+            }
+         }
+         
+      } while(anyCellNeedsTracing);
+} // pragma omp parallel
+      phiprof::stop("loop");
+      
+      bool redWarning = false;
+      MPI_Allreduce(&warnMaxStepsExceeded, &redWarning, 1, MPI_C_BOOL, MPI_LOR, MPI_COMM_WORLD);
+      
+      if(redWarning && rank == MASTER_RANK) {
+         logFile << "Warning: reached the maximum number of tracing steps " << maxTracingSteps << " allowed for full-box ionosphere tracing." << endl;
+      }
+      
+      std::vector<int> reducedCellFWConnection(globalDccrgSize);
+      std::vector<int> reducedCellBWConnection(globalDccrgSize);
+      MPI_Allreduce(cellFWConnection.data(), reducedCellFWConnection.data(), globalDccrgSize, MPI_INT, MPI_MIN, MPI_COMM_WORLD);
+      MPI_Allreduce(cellBWConnection.data(), reducedCellBWConnection.data(), globalDccrgSize, MPI_INT, MPI_MIN, MPI_COMM_WORLD);
+      // Tracing: 0 not completed, 1 out of the box, 2 ionosphere
+      // 0: closed
+      // 1: closed-open
+      // 2: open-closed   
+      // 3: open-open
+      // 4: closed-loop
+      // 5: loop-closed
+      // 6: open-loop
+      // 7: loop-open
+      // 8: loop-loop
+      // 9: invalid
```

Comment:
This should also really be an enum, instad of an int with magical values that happen to be commented somewhere in the middle of a function

Reply:
Well, sure, can be, it's still only going to be integers in the output file though.

Reply:
enums implemented an hour ago.